intellij/src/saros/intellij/ui/tree/SessionTreeRootNode.java

```diff
@@ -101,6 +116,21 @@ public SessionTreeRootNode(SessionAndContactsTreeView treeView) {
     sessionManager.addSessionLifecycleListener(sessionLifecycleListener);
   }
 
+  @Override
+  public void dispose() {
+    sessionManager.removeSessionLifecycleListener(sessionLifecycleListener);
+
+    ISarosSession currentSession = session;
+
+    if (currentSession != null) {
+      currentSession.removeListener(sessionListener);
+      removeSessionNode(currentSession);
+    }
+
+    sessionNodeList.clear();
```

Comment:
The job of a dispose method is either to free unmanaged resources / memory and to deregister stuff properly so that the object can be garbage collected. Of course you can set all references to null / clear datastructures etc. to help the garbage collector but this is just boiler plate code with little benefits.

Reply:
Removed. Just un-registering the listeners now.